11x-new-kafka-roller.md

```diff
@@ -0,0 +1,263 @@
+# Kafka Roller 2.0
+
+## Current situation
+
+The Kafka Roller is an internal Cluster Operator component that's responsible for coordinating the rolling restart or reconfiguration of Kafka pods when:
+- non-dynamic reconfigurations needs to be applied
+- update in Kafka CR is detected
+- a TLS certificate is renewed
+- pods have been [manually annotated](https://strimzi.io/docs/operators/latest/full/deploying#rolling_pods_manually_alternative_to_drain_cleaner) by the user for controlled restarts
+- pod is stuck and has a pending update (e.g. not running with the desired version or configuration)
+- Kafka broker is unresponsive to Kafka Admin connections
+
+A pod is considered stuck if it is in one of following states:
+- `CrashLoopBackOff`
+- `ImagePullBackOff`
+- `ContainerCreating`
+- `Pending` and `Unschedulable`
+
```

Comment:
I wonder if we are missing some details here for those who aren't as familiar with the current KafkaRoller. Based on points we have elsewhere I wonder if we should include something like:
```
Some relevant internal details of the existing KafkaRoller are:
- The KafkaRoller processes one node at a time.
- If a node cannot be processed (e.g. there was a failure while reconfiguring or restarting) the KafkaRoller will process the node again after a delay.
- For each node the KafkaRoller will try to process it up to 10 times before failing the reconciliation (this number is hard-coded).
```